frontend-react/src/components/Admin/DisplayMeta.tsx

```diff
@@ -1,24 +1,34 @@
 import { useEffect, useState } from "react";
 
-import { MetaData } from "../../resources/OrgSettingsBaseResource";
+import { MetaTaggedResource } from "../../resources/OrgSettingsBaseResource";
 import { formatDate } from "../../utils/misc";
 
-type Props = {
-    metaObj?: MetaData | undefined;
+type DisplayMetaProps = {
+    metaObj?: MetaTaggedResource;
 };
 
-export const DisplayMeta = (props: Props) => {
-    const [metaData, setMetaData] = useState<MetaData>();
+export const DisplayMeta = ({ metaObj }: DisplayMetaProps) => {
+    const [metaData, setMetaData] = useState<MetaTaggedResource>();
 
     useEffect(() => {
-        if (!props.metaObj) {
+        if (!metaObj) {
             return;
         }
-        setMetaData(props.metaObj);
-    }, [props]);
+        setMetaData(metaObj);
+    }, [metaObj]);
 
-    return metaData ? (
-        <>{`v${metaData.version} [${formatDate(metaData.createdAt)}] 
-        ${metaData.createdBy}`}</>
-    ) : null;
+    if (!metaData) {
```

Comment:
Nitpick, but can this live down with the other return? Components don't benefit from an early return so all the code below runs anyway. This made me think we were conditionally performing what was underneath it at first

Reply:
yeah, could take this out and default metaData to an empty object
```suggestion
    const { version, createdAt, createdBy } = {}  = metaData;
```
or better, yet default the state `metaData` to an empty object from the start.

However, I'm confused about the idea of the early return not working. I did a quick smoke test on this idea, and it looks like the function is exiting on line 21 when `metaData` comes through undefined. Can you clarify what you mean here? feel like I may be missing something important...

Reply:
There's a sneaky little one-liner at the _veeeery_ bottom of the page that explains it: https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

> Returning `null from a component’s `render` method does not affect the firing of the component’s lifecycle methods. For instance `componentDidUpdate` will still be called.

It's tricky because this uses class-based component verbiage, but essentially when a functional component is called, it's stepping through a setup lifecycle and I believe it runs through without returning during setup to initiate the hooks, and then renders based on the default state from hooks. Also, [here](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level), there's a great explanation lower on this page about why hooks _can't_ be conditionally rendered and why this lifecycle is not something you can return early from.

Reply:
Retracing what's happening in this, per the lifecycle, I think it's in this order:
- Lifecycle runs setup
- UseState will run
- UseEffect will run
- `metaData` is destructured
- `versionDisplay` is set
- ...
- Component mounted and ready to render
- Renders
- Exits at `null` if `!metaData`